frontend/src/metabase/static-viz/lib/numbers.ts

```diff
@@ -24,74 +25,13 @@ const DEFAULT_OPTIONS = {
 };
 
 export const formatNumber = (number: number, options?: NumberFormatOptions) => {
-  const {
-    number_style,
-    currency,
-    currency_style,
-    number_separators: [decimal_separator, grouping_separator],
-    decimals,
-    scale,
-    prefix,
-    suffix,
-    compact,
-  } = handleSmallNumberFormat(number, { ...DEFAULT_OPTIONS, ...options });
-
-  function createFormat(compact?: boolean) {
-    if (compact) {
-      return new Intl.NumberFormat("en", {
-        style: number_style !== "scientific" ? number_style : "decimal",
-        notation: "compact",
-        compactDisplay: "short",
-        currency: currency,
-        currencyDisplay: currency_style,
-        useGrouping: true,
-        maximumFractionDigits: decimals != null ? decimals : 2,
-      });
-    }
-
-    return new Intl.NumberFormat("en", {
-      style: number_style !== "scientific" ? number_style : "decimal",
-      notation: number_style !== "scientific" ? "standard" : "scientific",
-      currency: currency,
-      currencyDisplay: currency_style,
-      useGrouping: true,
-      minimumFractionDigits: decimals,
-      maximumFractionDigits: decimals != null ? decimals : 2,
-    });
-  }
-
-  const format = createFormat(compact);
-
-  const separatorMap = {
-    ",": grouping_separator || "",
-    ".": decimal_separator,
+  const optionsWithDefault = {
+    ...DEFAULT_OPTIONS,
+    ...options,
```

Comment:
The [previous attempt](https://github.com/metabase/metabase/blob/c155666ca52f8a91146fe78bb861ebe9c1c59c2d/frontend/src/metabase/static-viz/lib/numbers.ts#L28-L30) failed because I didn't pass default options when the `options` parameter is `null` causing the error.